src/js/utils/MetronomeUtil.js

```diff
@@ -91,5 +91,36 @@ module.exports = {
     });
 
     return rootTree;
+  },
+
+  parseJob(job) {
+    let {history}= job;
+
+    if (history == null) {
+
+      return job;
+    }
+
+    let {failedFinishedRuns = [], successfulFinishedRuns = []} = history;
+
+    failedFinishedRuns = failedFinishedRuns.map(function (jobRun) {
+
```

Comment:
I know, it's ton of work, and I really appreciate your work! I didn't added these lines to offend you. I thought we're supposed to, which is due to the lack of a defined set of rules.